src/snowflake/snowpark/modin/plugin/_internal/index.py

```diff
@@ -82,11 +45,52 @@ def __init__(
         copy: bool = False,
         name: object = None,
         tupleize_cols: bool = True,
+        convert_to_lazy: bool = True,
     ) -> None:
+        """
+        Immutable sequence used for indexing and alignment.
+
+        The basic object storing axis labels for all pandas objects.
+
+        Parameters
+        ----------
+        data : array-like (1-dimensional)
+        dtype : str, numpy.dtype, or ExtensionDtype, optional
+            Data type for the output Index. If not specified, this will be
+            inferred from `data`.
+            See the :ref:`user guide <basics.dtypes>` for more usages.
+        copy : bool, default False
+            Copy input data.
+        name : object
+            Name to be stored in the index.
+        tupleize_cols : bool (default: True)
+            When True, attempt to create a MultiIndex if possible.
+        convert_to_lazy : bool (default: True)
+            When True, create a lazy index object, otherwise, create an index object that locally saves a pandas index
```

Comment:
When True, create a lazy index object from a local `data` input, otherwise, create an index object that saves a pandas index locally.
We only set convert_to_lazy as False to avoid pulling data back and forth from Snowflake, e.g., when calling df.columns, the column data should always be kept locally.